students/ScottL/session08/circle_class.py

```diff
@@ -0,0 +1,84 @@
+
+"""
+Homework 8 - Circle Class
+Use objects and their associated methods for characterizing circles and spheres.
+"""
+
+import math
+
+
+class Circle:
+    """Define and quantify properties for circles with a given radius"""
+
+    def __init__(self, radius):
+        self.radius = radius
+
+    @property
+    def diameter(self):
+        """Return the diameter calculated from the radius property."""
+        return self.radius * 2
+
+    @diameter.setter
+    def diameter(self, value):
+        self.radius = value / 2
+
+    @property
+    def area(self):
+        """Return the area of a circle calculated from the radius property."""
+        return math.pi * self.radius ** 2
+
+    @classmethod
+    def from_diameter(cls, diameter):
+        """Define the object and its radius based an input for the diameter."""
+        return cls(diameter / 2)
+
+    def __str__(self):
+        return f"Circle with a radius of: {self.radius}"
+
+    def __repr__(self):
+        return f"Circle({self.radius})"
+
+    def __add__(self, other):
+        return f"Circle({self.radius + other.radius})"
```

Comment:
This should result in a `Circle()` object, not a string value

Reply:
ahh yeah that makes more sense, i'll get that adjusted